### PR TITLE
Add scenario test for 1.5.2 

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
Add scenario test for 1.5.2.7, 1.5.2.8 and 1.5.2.9

### CODE REVIEW

File: product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java [hunk 1]
```diff
@@ -1052,7 +1052,33 @@ public HttpResponse getAPIListFromStoreAsAnonymousUser(String tenantDomain)
...
+     *
+     * @return HttpResponse - Response with tags of APIs visible to anonymous users
+     * @throws APIManagerIntegrationTestException
+     */
+    public HttpResponse getTagListFromStoreAsAnonymousUser()
+            throws APIManagerIntegrationTestException {
+        try {
+            HttpResponse httpResponse = HttpClient.doPost(
+                    new URL(backendURL + "/site/blocks/tag/tag-cloud/ajax/list.jag?action=getAllTags" +
+                            "&tenant=carbon.super"),
```
Comment (isharac) [inline product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java]: yiu can merge with the line below

File: product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java [hunk 2]
```diff
@@ -1052,7 +1052,33 @@ public HttpResponse getAPIListFromStoreAsAnonymousUser(String tenantDomain)
...
+                    "", requestHeaders);
+
+            if (new JSONObject(httpResponse.getData()).getBoolean("error")) {
+                throw new APIManagerIntegrationTestException("Error when getting tag list as anonymous user");
```
Comment (isharac) [inline product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java]: let's not throw APIManagerIntegrationTestException instead just throw Exception

File: product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java (hunk 1 above)
Comment (ruwiniwj) [reply product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java]: Fixed

File: product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java (hunk 2 above)
Comment (ruwiniwj) [reply product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java]: Fixed

File: product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java [hunk 3]
```diff
@@ -1052,7 +1052,30 @@ public HttpResponse getAPIListFromStoreAsAnonymousUser(String tenantDomain)
         }
     }
 
+    /**
+     * Get tags of APIs in Store for an anonymous user
+     *
+     * @return HttpResponse - Response with tags of APIs visible to anonymous users
+     * @throws APIManagerIntegrationTestException
```
Comment (isharac) [inline product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java]: change the java doc as well

Comment (ruwiniwj) [reply product-scenarios/scenarios-common/src/main/java/org/wso2/am/scenario/test/common/APIStoreRestClient.java]: Fixed

File: product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java [hunk 4]
```diff
@@ -103,38 +111,159 @@ public void testVisibilityOfAPISRestrictedByMultipleRoles() throws Exception {
...
-        getAPI();
-        publishAPI(apiName, ADMIN_LOGIN_USERNAME);
+        createAPI(apiRequest);
+        getAPI(PUBLISHER_CREATOR_USERNAME);
+        publishAPI(apiName, PUBLISHER_CREATOR_USERNAME);
         loginToStore(userName, password);
         isAPIVisibleInStore(apiName, apiStoreClient);
     }
 
+//    todo need to config a secondary userstore to check roles with spaces
```
Comment (ChamodDamitha) [inline product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: Shall we replace these commented codes if the todo is done

Comment (ruwiniwj) [reply product-scenarios/1-manage-public-partner-private-apis/1.5-expose-api-to-the-app-developer/1.5.2-expose-api-to-subset-of-app-developer-restricted-by-roles/src/test/java/org/wso2/am/scenario/tests/rest/api/restrictedVisibility/RestApiVisibilityRestrictedByRolesTestCase.java]: The requirement is not implemented yet

**Pull request merged**